spek-junit-platform-engine/src/test/kotlin/org/jetbrains/spek/engine/LifecycleListenerOrderTest.kt

```diff
@@ -72,12 +72,12 @@ class LifecycleListenerOrderTest: AbstractSpekTestEngineTest() {
         executeTestsForClass(TestSpek::class)
         // afterXXX fixtures are executed from bottom to top
         val expected = """
-        1
         3
         2
         1
         3
         2
+        1
```

Comment:
This would affect users using memoized, subjects and the ones registering `LifecycleListener`s directly. Given that this change is the **correct** behaviour for subjects and memoized, the only thing left to check is the usage for `LifecycleListener`s .